### ISSUE TITLE
1.14 - Local variables disappear in the same trigger

### ISSUE THREAD

**Issue by antoinech2** [SkriptLang/Skript#2316]
### Description
Hello, I'm using the skript-bd add-on to be able to use MySql requests in Skript.
But since Skript 2.4 in 1.14, the local variables disappear when I use a skript-bd's syntax.
This is a Skript issue, because it work fine in Skript 2.2 in 1.12.2. I juste updated Skript and it does not work anymore!
Furthermore, I reported this bug to @btk5h, the skript-bd addon developer. He told me that it is a Skript issue.

I think it is related to thread desynchronization but I don't know myself.

### Steps to Reproduce
```	set {_test} to "test"
	send "BEFORE: %{_test}%" to player
	execute unsafe "SELECT log_console,log_ingame,log_script,log_global FROM skript_log_config WHERE script = '%{_script}%'" in {mysql_db} and store the result in {_mysql_result::*}
# The bug works whatever the request is
	send "AFTER: %{_test}%" to player
```

### Expected Behavior
The local value {_test} still has its value, "test" after the SQL request.

### Errors / Screenshots
After the request, the value of test {_test} variable is deleted, as you can see:
![image](https://user-images.githubusercontent.com/42295406/62124914-c97c9100-b2cb-11e9-9ad0-ee8ecfaa802e.png)

### Server Information
* **Server version/platform:**       PaperSpigot 1.14.4 # 152
* **Skript version:**                Skript 2.4-beta4 (bug only in 2.4 versions, work in 2.2 versions)
* **skript-db version:**          0.1.1 or 0.2.1 (bug in all versions with Skript 2.4)

### Additional Context
Please fix this bug because it break all my scripts with these syntax.

**Comment by TheLimeGlass** [comment]
Skript is not designed to work asynchronously from the main thread. Skungee has the same issue.
Once the event changes/removes the variables get deleted from the currentEvent as it's a local variable and only present on the event.

**Comment by TheLimeGlass** [comment]
https://github.com/SkQuery/SkQuery/issues/8

**Comment by Blueyescat** [comment]
Yes it is new and started after https://github.com/SkriptLang/Skript/commit/237b3dd928f3108d154204d3e98af1208e7679f4. Skript devs know about this, addons has to fix that issue themselves. See https://github.com/btk5h/skript-db/pull/14

**Comment by antoinech2** [comment]
Thank you very much for your quick response,

If I understand correctly it is skript-bd who must solve the problem on his side?
And maybe this fix can work (btk5h/skript-db#14)? (It had been approved by the dev but never been put in place)

**Comment by bensku** [comment]
Kind of duplicate of #1817. This pretty much needs a new variable system to solve.
